domain/secret/service/delete.go

```diff
@@ -17,7 +17,12 @@ import (
 
 // DeleteObsoleteUserSecretRevisions deletes any obsolete user secret revisions that are marked as auto-prune.
 func (s *SecretService) DeleteObsoleteUserSecretRevisions(ctx context.Context) error {
-	return s.st.DeleteObsoleteUserSecretRevisions(ctx)
+	deletedRevisionIDs, err := s.secretState.DeleteObsoleteUserSecretRevisions(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = s.secretBackendReferenceMutator.RemoveSecretBackendReference(ctx, deletedRevisionIDs...)
+	return errors.Trace(err)
```

Comment:
We shouldn't error here. As far as the caller is concerned, the revision is deleted. It's more a juju problem the ref is still there. It doesn't affect the core semantic of the api. And we will be implementing a reconciliation strategy. I think logging the error is sufficient.